### ISSUE TITLE
Focus trap issues within Web Components

### ISSUE THREAD

**Issue by sbutler2901** [carbon-design-system/ibm-security#700]
## Bug - Focus trap functionality breaks when rendered within a web component

**Expected behavior -**

When a component that utilizes the FocusTrap is used within a web component the user will be able to interact with the elements rendered within the focus trap and only those elements. Tabbing should be locked so that the selected element cannot escape the trap either.

**Actual behavior -**

The focus trap blocks interactions altogether including clicks and accessibility clicks (space / enter key). The user must press the escape button which essentially remove these focus trap and then the element is interactable by the user. This removes the benefits of the focus trap, such as tabbing outside of the intended trap elements.
